build.sbt

```diff
@@ -99,16 +101,21 @@ lazy val client = libraryProject("client")
     scalaXml % "test->compile")
 
 lazy val clientMetrics = libraryProject("client-metrics")
+  .settings(
```

Comment:
This one doesn't add any new dependencies.  I think this could live directly in the client project, with the implementations in their own modules.

Reply:
Makes sense yes.